provider/kubernetes/kubernetes.go

```diff
@@ -454,18 +458,23 @@ func getTLS(ingress *extensionsv1beta1.Ingress, k8sClient Client) ([]*tls.Config
 	return tlsConfigs, nil
 }
 
-func endpointPortNumber(servicePort corev1.ServicePort, endpointPorts []corev1.EndpointPort) int {
-	if len(endpointPorts) > 0 {
-		// name is optional if there is only one port
-		port := endpointPorts[0]
-		for _, endpointPort := range endpointPorts {
-			if servicePort.Name == endpointPort.Name {
-				port = endpointPort
-			}
+// endpointPortNumber returns the port to be used for this endpoint. It is zero
+// if the endpoint does not match the given service port.
+func endpointPortNumber(servicePort corev1.ServicePort, endpointPorts []corev1.EndpointPort) int32 {
+	// Is this reasonable to assume?
+	if len(endpointPorts) == 0 {
```

Comment:
Are you checking here to see if `endpointPorts` is defined at all? or just that the struct is empty?

Reply:
It'd be the same as both a `nil` slice and a non-nil, empty slice return `true` for a length check against zero.

Again, I didn't make any semantical changes to this part of the logic; I only improved readability so that we return right away if the length is zero.